imap/index.c

```diff
@@ -1211,11 +1211,11 @@ EXPORTED int index_fetch(struct index_state *state,
     r = index_lock(state, readonly);
     if (r) return r;
 
-    if (!strcmp("$", sequence)) {
-        seq = state->searchres;
+    seq = _parse_sequence(state, sequence, usinguid);
+    if (!strcmpsafe("$", sequence)) {
         usinguid = 1;
```

Comment:
After we've called _parse_sequence, we're setting usinguid to 1. But _parse_sequence has different behaviour depending on usinguid. Does that matter?

Reply:
It shouldn't, as `_parse_sequence()` will return immediately if `sequence == "$"`.  I don't believe this changes the existing behavior and the Cass tests still pass, assuming that we have proper coverage.